src/dh/clu_dh.c

```diff
@@ -0,0 +1,393 @@
+                inSz = pDer->length;
+                XMEMCPY(in, pDer->buffer, pDer->length);
+            }
+
+            if (ret == WOLFCLU_SUCCESS &&
+                    wc_DhKeyDecode(in, &idx, &dh, inSz) != 0) {
+                WOLFCLU_LOG(WOLFCLU_E0, "Unable to decode input params");
+                ret = WOLFCLU_FATAL_ERROR;
+            }
+
+            if (in != NULL)
+                XFREE(in, NULL, DYNAMIC_TYPE_TMP_BUFFER);
+            if (pDer != NULL)
+                wc_FreeDer(&pDer);
+        }
+    }
+
+    if (ret == WOLFCLU_SUCCESS) {
+        if (out != NULL) {
+            bioOut = wolfSSL_BIO_new_file(out, "wb");
+            if (bioOut == NULL) {
+                WOLFCLU_LOG(WOLFCLU_E0, "Unable to open output file %s",
+                        optarg);
+                ret = WOLFCLU_FATAL_ERROR;
+            }
+        }
+        else {
+            /* use stdout by default */
+            bioOut = wolfSSL_BIO_new(wolfSSL_BIO_s_file());
+            if (bioOut == NULL) {
+                ret = WOLFCLU_FATAL_ERROR;
+            }
+            else {
+                if (wolfSSL_BIO_set_fp(bioOut, stdout, BIO_NOCLOSE)
+                        != WOLFSSL_SUCCESS) {
+                    ret = WOLFCLU_FATAL_ERROR;
+                }
+            }
+        }
+    }
+
+    /* generate the dh parameters */
+    if (ret == WOLFCLU_SUCCESS && bioIn == NULL) {
+        if (wc_DhGenerateParams(&rng, modSz, &dh) != 0) {
+            WOLFCLU_LOG(WOLFCLU_E0, "Error generating parameters");
+            ret = WOLFCLU_FATAL_ERROR;
+        }
+    }
+
+    /* print out the dh parameters */
+    if (ret == WOLFCLU_SUCCESS && !noOut) {
+        byte* outBuf = NULL;
+        byte* pem    = NULL;
+        word32 outBufSz = 0;
+        word32 pemSz    = 0;
+
+        if (wc_DhParamsToDer(&dh, NULL, &outBufSz) != LENGTH_ONLY_E) {
+            WOLFCLU_LOG(WOLFCLU_E0, "Unable to get output buffer size");
+            ret = WOLFCLU_FATAL_ERROR;
+        }
+
+        if (ret == WOLFCLU_SUCCESS) {
+            outBuf = (byte*)XMALLOC(outBufSz, NULL, DYNAMIC_TYPE_TMP_BUFFER);
+            if (outBuf == NULL) {
+                ret = WOLFCLU_FATAL_ERROR;
+            }
+        }
+
+        if (ret == WOLFCLU_SUCCESS &&
+                wc_DhParamsToDer(&dh, outBuf, &outBufSz) <= 0) {
+            ret = WOLFCLU_FATAL_ERROR;
+        }
+
+        if (ret == WOLFCLU_SUCCESS) {
+            pemSz = wc_DerToPem(outBuf, outBufSz, NULL, 0, DH_PARAM_TYPE);
+            if (pemSz > 0) {
+                pem = (byte*)XMALLOC(pemSz, NULL, DYNAMIC_TYPE_TMP_BUFFER);
+                if (pem == NULL) {
+                    ret = WOLFCLU_FATAL_ERROR;
+                }
+            }
+            else {
+                ret = WOLFCLU_FATAL_ERROR;
+            }
+        }
+
+        if (ret == WOLFCLU_SUCCESS) {
+            pemSz = wc_DerToPem(outBuf, outBufSz, pem, pemSz, DH_PARAM_TYPE);
+            if (pemSz <= 0) {
```

Comment:
pemSz is a word32 type, we should use an int if checking if it will be less than 0 here, looks like the code from clu_dsa.c also has this issue. Can you please fix both locations

Reply:
Done